js/app/dashboard/redux/taskListEntityReducers.js

```diff
@@ -0,0 +1,106 @@
+import {
+  MODIFY_TASK_LIST_REQUEST,
+  MODIFY_TASK_LIST_REQUEST_SUCCESS,
+  TASK_LIST_UPDATED,
+  UPDATE_TASK
+} from "./actions";
+import {
+  taskUtils,
+  taskListUtils,
+  taskListEntityUtils,
+} from '../../coopcycle-frontend-js/logistics/redux'
+
+const initialState = {
+  byId: {}
+}
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case MODIFY_TASK_LIST_REQUEST: {
+      let entity = taskListEntityUtils.findTaskListByUsername(state.byId, action.username)
+
+      if (entity === undefined) {
+        // eslint-disable-next-line no-console
+        console.assert(false, `entity is undefined: username: ${action.username}`)
+
+        return state
+      }
+
+      let newEntity = {
+        ...entity,
+        itemIds: taskUtils.tasksToIds(action.tasks),
+      }
+
+      let newItems = taskListEntityUtils.addOrReplaceTaskList(state.byId, newEntity)
+
+      return {
+        ...state,
+        byId: newItems,
+      }
+    }
+    case MODIFY_TASK_LIST_REQUEST_SUCCESS: {
+      let newEntity = taskListUtils.replaceTasksWithIds(action.taskList)
+      let newItems = taskListEntityUtils.addOrReplaceTaskList(state.byId, newEntity)
+
+      return {
+        ...state,
+        byId: newItems,
+      }
+    }
+    case TASK_LIST_UPDATED: {
+      let entityByUsername = taskListEntityUtils.findTaskListByUsername(state.byId, action.taskList['username'])
+
+      if (entityByUsername === undefined) {
+        return state
+      }
+
+      // items: [
+      //   {
+      //     task: '/api/tasks/21',
+      //     position: 0
+      //   },
+      //   {
+      //     task: '/api/tasks/18',
+      //     position: 1
+      //   }
+      // ],
+      let taskCollectionItems = action.taskList.items
+      let itemIds = taskCollectionItems.map(item => item.task)
```

Comment:
I think there's a problem here. The previous code was like this:

```es6
case TASK_LIST_UPDATED:

  taskListIndex = _.findIndex(state.taskLists, taskList => taskList['@id'] === action.taskList['@id'])

  if (-1 === taskListIndex) {

    return state
  }

  newTaskLists.splice(taskListIndex, 1, {
    ...state.taskLists[taskListIndex],
    distance: action.taskList.distance,
    duration: action.taskList.duration,
    polyline: action.taskList.polyline,
  })

  return {
    ...state,
    taskLists: newTaskLists,
  }
```
  
i.e it was **NOT** touching the items. 

This event is generated by an async job, which recalculates the distance/duration of the items in the list.
It means that depending on the order of job scheduling, which is unpredictable, we may be receiving an event for a **previous version** of the list (with other tasks). 

I think this is was caused problems with tasks being "visually" unassigned.